spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestDropTable.java

```diff
@@ -0,0 +1,152 @@
+    dropTableInternal();
+  }
+
+  private void dropTableInternal() {
+    assertEquals("Should have expected rows",
+        ImmutableList.of(row(1, "test")), sql("SELECT * FROM %s", tableName));
+
+    List<String> previousDataFiles = getDataOrMetadataFiles(true);
+    Assert.assertTrue("The number of data files should be larger than zero", previousDataFiles.size() > 0);
+
+    sql("DROP TABLE %s", tableName);
+    Assert.assertFalse("Table should not exist", validationCatalog.tableExists(tableIdent));
+
+    if (catalogName.equals("testhadoop")) {
+      // HadoopCatalog drop table without purge will delete the base table location.
+      Assert.assertEquals("The number of data files should be zero", 0, getDataOrMetadataFiles(true).size());
+      Assert.assertEquals("The number of metadata files should be zero", 0, getDataOrMetadataFiles(false).size());
+    } else {
+      Assert.assertEquals("The data files should not change", previousDataFiles, getDataOrMetadataFiles(true));
+    }
+  }
+
+  @Test
+  public void testPurgeTable() {
+    assertEquals("Should have expected rows",
+        ImmutableList.of(row(1, "test")),
+        sql("SELECT * FROM %s", tableName));
+
+    List<String> previousDataFiles = getDataOrMetadataFiles(true);
+    Assert.assertTrue("The number of data files should be larger than zero", previousDataFiles.size() > 0);
+
+    sql("DROP TABLE %s PURGE", tableName);
+    Assert.assertFalse("Table should not exist", validationCatalog.tableExists(tableIdent));
+    Assert.assertEquals("The number of data files should be zero", 0, getDataOrMetadataFiles(true).size());
+    Assert.assertEquals("The number of metadata files should be zero", 0, getDataOrMetadataFiles(false).size());
+  }
+
+  @Test
+  public void testPurgeTableGCDisabled() {
+    sql("ALTER TABLE %s SET TBLPROPERTIES (gc.enabled = false)", tableName);
+    List<String> previousDataFiles = getDataOrMetadataFiles(true);
+    List<String> previousMetadataFiles = getDataOrMetadataFiles(false);
+    Assert.assertTrue("The number of data files should be larger than zero", previousDataFiles.size() > 0);
+    Assert.assertTrue("The number of metadata files should be larger than zero", previousMetadataFiles.size() > 0);
+
+    AssertHelpers.assertThrows("Purge table is not allowed when GC is disabled", ValidationException.class,
+        "Cannot purge table: GC is disabled (deleting files may corrupt other tables",
+        () -> sql("DROP TABLE %s PURGE", tableName));
+
+    Assert.assertTrue("Table should not been dropped", validationCatalog.tableExists(tableIdent));
+    Assert.assertEquals("The data files should not change", previousDataFiles, getDataOrMetadataFiles(true));
+    Assert.assertEquals("The metadata files should not change", previousMetadataFiles, getDataOrMetadataFiles(false));
+  }
+
+  private List<String> getDataOrMetadataFiles(boolean dataFiles) {
+    String baseLocation = getTableBaseLocation();
```

Comment:
one little thing you could do here to avoid relying on the actual location of the table and all that, is just read the path column out of the MANIFESTS and FILES metadata tables. That would remove some fo the logic in getTableBaseLocation.

See

https://iceberg.apache.org/docs/latest/spark-queries/#files
and
https://iceberg.apache.org/docs/latest/spark-queries/#manifests

Reply:
Changed to verify the files of `MANIFESTS` and `FILES`.